src/Client.php

```diff
@@ -278,19 +319,14 @@ public function increment($metrics, $delta = 1, $sampleRate = 1.0, array $tags =
     {
         $metrics = is_array($metrics) ? $metrics : [$metrics];
 
-        $data = [];
-        if ($sampleRate < 1.0) {
+        if ($this->isSampled($sampleRate, $postfix)) {
+            $data = [];
             foreach ($metrics as $metric) {
-                if ((mt_rand() / mt_getrandmax()) <= $sampleRate) {
-                    $data[$metric] = $delta . '|c|@' . $sampleRate;
-                }
-            }
-        } else {
-            foreach ($metrics as $metric) {
-                $data[$metric] = $delta . '|c';
+                $data[$metric] = $delta . '|c' . $postfix;
             }
+            return $this->send($data, $tags);
         }
-        return $this->send($data, $tags);
+        return $this;
```

Comment:
because when this value is not to be sent when being sampled, it shouldn't attempt to send anything. The `send` method returns `$this` anyway.